webpack.config.js

```diff
@@ -0,0 +1,69 @@
+var webpack = require('webpack');
+var path = require('path');
+var fs = require('fs');
+
+var externalModules = [
+	'mkfifo',
+	'sqlite3',
+	'mysql2',
+	'pg',
+	'mariasql',
+	'mssql',
+	'mysql',
+	'strong-oracle',
+	'oracle',
+	'oracledb',
+	'pg-query-stream'
+]
+
+// We iterate over the node modules and mark all optional dependencies as external
+var dirs = fs.readdirSync('./node_modules')
+for (let dir of dirs){
+	try {
+		let packageJson = JSON.parse(fs.readFileSync('./node_modules/' + dir + '/package.json'))
+		if (packageJson.optionalDependencies != null){
+			for (let key in packageJson.optionalDependencies) {
+				externalModules.push(new RegExp('^' + key + '(/.*)?$'))
+			}
+		}
+	} catch (e) {}
+}
+
+console.log('Using the following dependencies as external:', externalModules);
+
+module.exports = {
+	entry: './src/app.coffee',
+	output: {
+		filename: 'app.js',
+		path: path.resolve(__dirname, 'dist')
+	},
+	resolve: {
+		extensions: [".js", ".json", ".coffee"]
+	},
+	target: 'node',
+	module: {
+		rules: [
+			{
+				test: /\.js$/,
+				use: require.resolve('./remove-hashbang-loader')
+			},
+			{
+				test: /\.coffee$/,
+				use: require.resolve('coffee-loader')
+			}
+		]
+	},
+	externals: (context, request, callback) => {
+		for (let m of externalModules) {
+			if ((typeof m === 'string' && m === request) || (m instanceof RegExp && m.test(request))) {
+				return callback(null, 'commonjs ' + request);
+			}
```

Comment:
It might be worth adding an else that throws an error, rather than silently swallowing invalid `externalModules` entries